### PR TITLE
Merge global refactoring to Master

### PR DESCRIPTION
**Pull request by Xen0Xys**
The objective of this PR is to suggest a way to refactor the project and mainly the Javascript part with some improvements :
- [x] Splitting into multiple functions and objects
- [x] Uniform naming conventions
- [x] Optional and needed dependencies for Python (like mocpy)
- [x] Improvement to some Python functions (like add_table)
- [x] Uniformed documentation with jsdoc and docstring
- [x] Minimal supported python version in `pyproject.toml`
- [x] Change add_listener function to set_listener in Python
- [x] #77

### CODE REVIEW

**Comment by ManonMarchand** [comment]
Maybe we could run the tests for the oldest and newest python versions that are supported ? (3.8 and 3.12)

**Comment by ManonMarchand** [comment]
Fixes #77

File: js/imports.js
Comment (bmatthieu3) [inline js/imports.js:1]: I would have call it maybe aladin_lite.js or something suggesting that it refers to aladin lite.

File: js/models/message_handler.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  handleAddCatalogFromURL(msg) {
+    const options = convertOptionNamesToCamelCase(msg["options"] || {});
+    this.aladin.addCatalog(A.catalogFromURL(msg["votable_URL"], options));
+  }
+
+  handleAddMOCFromURL(msg) {
+    const options = convertOptionNamesToCamelCase(msg["options"] || {});
+    if (options["lineWidth"] === undefined) options["lineWidth"] = 3;
```
Comment (bmatthieu3) [inline js/models/message_handler.js:24]: I do not think it is needed anymore. In aladin lite there is a default linewidth given if you do not explicitely give one. The default is indeed 3

File: js/models/message_handler.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    this.aladin.addMOC(A.MOCFromURL(msg["moc_URL"], options));
+  }
+
+  handleAddMOCFromDict(msg) {
+    const options = convertOptionNamesToCamelCase(msg["options"] || {});
+    if (options["lineWidth"] === undefined) options["lineWidth"] = 3;
```
Comment (bmatthieu3) [inline js/models/message_handler.js:30]: same here

**Review by bmatthieu3 — CHANGES_REQUESTED** [review]
Just little comments on the JS part. Otherwise it is good to merge from my point of view.

**Comment by Xen0Xys** [comment]
Do not merge until the recursion bug isn't fixed !

File: .github/workflows/python-tests.yml [hunk 3]
```diff
@@ -11,13 +11,16 @@ on:
...
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8, 3.12]
     steps:
       - name: "Checkout branch"
         uses: actions/checkout@v4
       - name: "Set up Python on Ubuntu"
-        uses: actions/setup-python@v5
+        uses: actions/setup-python@v2
```
Comment (ManonMarchand) [inline .github/workflows/python-tests.yml:21]: why the downgrade?

File: src/ipyaladin/aladin.py [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+            fov = fov.deg
+        self._fov = fov
+        self.send({"event_name": "change_fov", "fov": fov})
+
+    @property
+    def target(self) -> SkyCoord:
+        """The target of the Aladin Lite widget.
+
+        It can be set with either a string of an `~astropy.coordinates.SkyCoord` object.
+
```
Comment (ManonMarchand) [inline src/ipyaladin/aladin.py:154]: typo: a string **or** an

File: src/ipyaladin/aladin.py [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+                   ]
+        >>> table = QTable([ra, dec, name],
+                            names=("ra", "dec", "name"),
+                            meta={"name": "my sample table"})
+        >>> aladin.add_table(table)
+        And the table should appear in the output of Cell 1!
+
+        """
+        # this library must be installed, and is used in votable operations
+        # http://www.astropy.org/
```
Comment (ManonMarchand) [inline src/ipyaladin/aladin.py:331]: I think we can remove the comment as astropy is in our mandatory dependencies now

File: src/ipyaladin/aladin.py [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+        self.send(
+            {
+                "event_name": "add_overlay_from_stcs",
+                "stc_string": stc_string,
+                "overlay_options": overlay_options,
+            }
+        )
+
+    # Note: the print() options end='\r'allow us to override the previous prints,
+    # thus only the last message will be displayed at the screen
```
Comment (ManonMarchand) [inline src/ipyaladin/aladin.py:360]: Don't know what the comment is referring to.

**Review by ManonMarchand — APPROVED** [review]

**Pull request merged**